### DISCUSSION TITLE
SGP.32 data type support

### DISCUSSION THREAD

**Discussion by Xzatrox** [esa/asn1scc#304] [General]
Hello, 

It's nice to connect.
I am searching for a library that supports SGP.32 data types such as Octet1, Octet16, and Octet32. I have confirmed that the library provides support for the generic Octet data type, but I am unsure about the rest. Additionally, I would appreciate any advice on how to quickly enhance the library by adding support for these data types and the entire SGP.32 specification.
Perhaps a more effective approach would be to wish for alternative implementations instead of improving this library. It is clear that nothing is universal in the world. Just trying to find the best way to reach the goal.
I would appreciate any valuable input and recommendations.

Thank you.

**Comment by maxime-esa** [comment]
Hello @Xzatrox 

`ASN1SCC`  is just the ASN.1 compiler. It is not shipped with libraries of ASN.1 types. 

It supports the most frequently used ASN.1 constructs, with the exception of `Object Information Classes` and type extensibility.
We can help you check if your grammar could be supported by ASN1SCC if you provide a pointer to the ASN.1 files.
